### PR TITLE
Add mbedtls 3.X support to core_pkcs11_mbedtls.c.

### PR DESCRIPTION
**Pull request by paulbartell**
Update CorePKCS11 mbedtls implementation to support mbedtls version 3.x in addition to version 2.x.

Run integration test with both mbedtls 3.1 and 2.28.

### CODE REVIEW

File: tools/mbedtls.cmake [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        INTERFACE ${mbedtls_2_SOURCE_DIR}/include/mbedtls
+    )
+
+    set_target_properties(
+        MbedTLS2_mbedcrypto MbedTLS2_mbedtls MbedTLS2_mbedx509
+        PROPERTIES ARCHIVE_OUTPUT_DIRECTORY ${MBEDTLS_2_BIN_DIR} LIBRARY_OUTPUT_DIRECTORY
+                                                                 ${MBEDTLS_2_BIN_DIR}
+    )
+
+    add_library(MbedTLS2::mbedtls ALIAS MbedTLS2_mbedtls)
```
Comment (archigup) [inline tools/mbedtls.cmake:40]: Do these libraries not need to be set if mbedtls is already populated? Or is the mbedtls_2_POPULATED check above not for checking if it was already downloaded?

File: .gitmodules [hunk 2]
```diff
@@ -1,12 +1,8 @@
...
-	update = none
 [submodule "source/dependency/3rdparty/mbedtls"]
-	path = source/dependency/3rdparty/mbedtls
-	url = https://github.com/ARMmbed/mbedtls.git
-	branch = mbedtls-2.16
-	update = none
+    path = source/dependency/3rdparty/mbedtls
+    url = https://github.com/ARMmbed/mbedtls.git
+    branch = v2.28.0
+    update = none
```
[the remark this answers is not in the document]
Comment (paulbartell) [reply .gitmodules:5]: I'll fixup the spacing. 
CBMC still depends on mbedtls headers and i'm not sure of the best way to handle that.

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -1,95 +1,74 @@
...
 
 # Add a target for running coverage on tests.
-add_custom_target( coverage
+add_custom_target(
+    coverage
     COMMAND ${CMAKE_COMMAND} -P ${MODULE_ROOT_DIR}/tools/cmock/coverage.cmake
-    DEPENDS cmock unity core_pkcs11_mbedtls_utest core_pkcs11_utest core_pki_utils_utest
+    DEPENDS cmock
+            unity
+            $<$<TARGET_EXISTS:core_pkcs11_mbedtls_utest>:core_pkcs11_mbedtls_utest>
```
[the remark this answers is not in the document]
Comment (paulbartell) [reply test/CMakeLists.txt:68]: It's cmake "generator syntax". It excludes dependencies from the coverage target when they are not enabled via the SYSTEM_TESTS or UNIT_TESTS defines.

File: tools/mbedtls.cmake (hunk 1 above)
Comment (paulbartell) [reply tools/mbedtls.cmake:40]: Looks like i'm missing a call to FetchContent_GetProperties here. Will add.

Comment (paulbartell) [reply tools/mbedtls.cmake:40]: mbedtls_2_POPULATED is not a cache variable, and is set by FetchContent_GetProperties and FetchContent_Populate.

**Comment by archigup** [comment]
Would be good to have readme section listing for dependencies listing mbedtls and which versions we support.
Other than that looks good to me

File: README.md [hunk 4]
```diff
@@ -26,42 +26,54 @@ The purpose of the corePKCS11 software only mock library is therefore to provide
...
+### Coverage Measurement and Instrumentation
+The following software is required to run the coverage target:
+- Linux, MacOS, or another POSIX-like environment.
+- A recent version of **GCC** or **Clang** with support for gcov-like coverage instrumentation.
+- **gcov** binary corresponding to your chosen compiler
+- **lcov** from the [Linux Test Project](https://github.com/linux-test-project/lcov)
+- **perl** needed to run the lcov utility.
 
-1. Go to the root directory of this repository. (Make sure that the **CMock** submodule is cloned as described [above](#checkout-cmock-submodule))
+Coverage builds are validated on recent versions of Ubuntu Linux.
```
Comment (archigup) [inline README.md:62]:
```suggestion
Coverage builds are validated on the latest Ubuntu LTS.
```

**Review by archigup — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**